### ISSUE TITLE
Look into implementing async/await

### ISSUE THREAD

**Issue by lassevk** [lassevk/Signals4Net#2]
It would be nice if tasks could be async, but this looks like a major undertaking.

In particular, there is particular concern regarding the system that detects which dependencies are in play if tasks get involved, and I have currently completely avoided problems with multiple threads in play.

Still, it would be nice.

As an example of what I could hope to accomplish:

    var context = new SignalContext();
    var employeeId = context.State(0);
    var employee = context.Computed(() => {
        await using var dbContext = await dbContextFactory.CreateAsync();
        return dbContext.Employees.FirstOrDefault(emp => emp.id == employeeId.Value);
    });

and then render `employee` in a blazor form.

This can currently be done in an effect:

    var context = new SignalContext();
    var employeeId = context.State(0);
    context.Effect(async () => {
        var id = employeeId.Value; // <-- this needs to be done before the first await
        await using var dbContext = await dbContextFactory.CreateAsync();
        return dbContext.Employees.FirstOrDefault(emp => emp.id == id);
    });

The main issue here is that the method will return upon the first encounter of awaiting a not-yet-complete object, and then dependency tracking will stop. In the above example the state variable is read before any such await, so this will be handled. However, you cannot create another compute that uses this employee to, say, calculate a nice display name, as this would run when the first await returned back to the signal system, not when the entire task has completed running.

Basically, in the above example, the signal system is oblivious to anything happening after the first await.

**Comment by lassevk** [comment]
Async code could also lead to multiple computed expressions being evaluated at the same time, or different parts of their code interleaved. This might make it hard to handle dependency detection properly. Maybe execution context could be used to handle this.

**Comment by lassevk** [comment]
I seem to have been able to implement async/await, but I need to decide if this is the way to go.

I particularly liked the simplicity of `state.Value`, but a property cannot be async, so once async/await was introduced I had to switch to methods.

This also posed a slight problem with Blazor, in terms of syntax, what seems the best syntax to express values to render is not possible since the markup is not async.

I will leave it on the attached branch for now, and think about it.
